cns/NetworkContainerContract.go

```diff
@@ -338,26 +340,47 @@ type CreateNetworkContainerResponse struct {
 	Response Response
 }
 
-// GetNetworkContainerStatusRequest specifies the details about the request to retrieve status of a specifc network container.
+// GetNetworkContainerStatusRequest specifies the details about the request to retrieve status of a specific network container.
 type GetNetworkContainerStatusRequest struct {
 	NetworkContainerid string
 }
 
-// GetNetworkContainerStatusResponse specifies response of retriving a network container status.
+// GetNetworkContainerStatusResponse specifies response of retrieving a network container status.
 type GetNetworkContainerStatusResponse struct {
 	NetworkContainerid string
 	Version            string
 	AzureHostVersion   string
 	Response           Response
 }
 
-// GetNetworkContainerRequest specifies the details about the request to retrieve a specifc network container.
+// GetAllNetworkContainersRequest specifies the details about the request to retrieve all network containers on CNS.
+type GetAllNetworkContainersRequest struct {
+	NodeID string
+}
+
+// GetAllNetworkContainersResponse returns all network containers' ID on the node.
+type GetAllNetworkContainersResponse struct {
+	NetworkContainerIDs []string
+	Response            Response
+}
+
+type ReplenishNetworkContainerRequest struct {
+	DncEP     string
+	InfraVnet string
+	NodeID    string
+}
```

Comment:
No, CNS does **not** reach out to DNC. The reason I use this pattern was to try to achieve the operation of `GET` the NCs details from DNC. Then I found out, as a matter of fact, this can be a `POST` on the same URL in the first API whose request body already has a payload containing the NCs that need to be updated. We can store all NCs without calling DNC.